Generate observations using mapping and csvData

pradh opened the pull request:
The main API returns observations as a map, associated with rows, so that the frontend can decide to link it to display rows.

There's also a helper to produce a text string from the map that's to be tweaked as necessary.

chejennifer commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,146 @@
+  const rowObs: RowObservations = new Map();
+  for (const idx of Array.from(csvData.rowsForDisplay.keys())) {
+    const row = csvData.rowsForDisplay.get(idx);
+    const obs = generateObservationsInRow(
```
nit: functions should be defined before they are used, so `generateObservationsInRow` should be defined before this current function

pradh replied:
I see. Moved it around.

chejennifer commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,146 @@
+    const hdrThing = obsGenMaps.valCol2Hdr.get(valColIdx);
+    const obs: Observation = new Map();
+    // TODO: Consider if we want to do some cleaning and checking for non-numeric value.
+    if (row[valColIdx] != null && row[valColIdx] !== "") {
```
can actually just use the lodash function `_.isEmpty()`: https://lodash.com/docs/4.17.15#isEmpty

pradh replied:
Nice tip!

chejennifer commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,146 @@
+
+// Helper maps used while generating observaitons.
+interface ObsGenMaps {
+  // Column Index -> mapping thing found in header for VALUE.
```
nit: "mapped thing found .."

pradh replied:
Done.

chejennifer reviewed (commented):
this is great!

pradh reviewed (commented):
Thanks for the review Jennifer!

jehangiramjad commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,148 @@
+  RowObservations,
+} from "../types";
+
+// Helper maps used while generating observaitons.
```
nit: typo at the end of this comment. s/observaitons/observations

pradh replied:
fixed.

jehangiramjad commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Given a mapping and CSV Data computes the observations for every row
+ * corresponding to |rowsForDisplay|.
```
nit: this function comment does not seem to follow the function comment format with @param and @returns etc

pradh replied:
Thanks for the reminder :)

jehangiramjad commented on wizard/src/utils/obs_generation.ts:
```diff
@@ -0,0 +1,148 @@
+}
+
+/**
+ * Given an Observation that was returned from generateRowObservations(), returns
```
ditto the nit comment about comment format above.

pradh replied:
Done.

jehangiramjad reviewed (approved):
Looks good! Just a couple of nitty comments.

pradh reviewed (commented):
Thank you for the comments Jehangir!

pradh commented:
@chejennifer: I have addressed your comments.  Going to submit this to unblock things next week.

Please add any further comments you have, and I will address them in a follow up PR...
